### DISCUSSION TITLE
Differentiate between local packages and cached ones

### DISCUSSION THREAD

**Discussion by EslamElHusseiny** [verdaccio/verdaccio#4983] [Q&A] [answered]
Hi, 
How can I differentiate between locally developed package and cached ones from nodejs.org proxy especially that both are stored in the same `storage` directory? 

Thanks!

**Comment by juanpicado** [comment]
There is a property on package access named storage and allows you to group by folders 

https://verdaccio.org/docs/packages

**Reply by juanpicado** [reply]
Ref https://github.com/verdaccio/verdaccio/issues/1383#issuecomment-509933674

**Reply by EslamElHusseiny** [reply]
I see what you mean. This is more of organising by having all the private packages under a specific scope(s) stored in one location and anything else "cached from proxies" stored in another location. 

Well, this isn't the case as our developers are allowed to publish with and without scope the private packages. 


And my questions is more about the already published and cached  npm packages in the same storage location. The situation is that now I have both the private and cached npm binaries stored in the same storage path and wanted to check if there is a way to know which is private and which is coming from the uplink.

**Reply by mbtools** [reply, marked as answer]
The `package.json` in the Verdaccio storage includes the uplink name and time stamp when it was fetched:

![image](https://github.com/user-attachments/assets/becfc817-0c0b-49e1-96e0-53692c825428)

**Reply by EslamElHusseiny** [reply]
@mbtools Thank you so much for your answer! 

I think this is a workable solution checking the `_uplinks` item for each package stored in `storage` directory.
